text/0000-location.md

```diff
@@ -0,0 +1,409 @@
+|Owner|org_id|The identifier of the organization responsible for maintaining the location. For GS1 locations, org_id is the Pike organization which claims ownership of the GS1 company prefix (via the "gs1_company_prefixes" field in the Pike records). The GLN will always start with the company prefix. The smart contract will check that the org claiming ownership lists the matching prefix in their Pike org.||
+
+
+## Transactions
+[transactions]: #transactions
+
+In order to add Locations to state, a transaction must be used. The following 
+transactions and their execution rules are designed for the Hyperledger Sawtooth
+platform and may differ for other transaction execution platforms.
+
+Locations are managed by submitting transactions to Hyperledger Grid, which will
+process them with the Grid Location smart contract. The following transactions 
+are supported:
+- LocationCreate - create a Location and store it in state.
+- LocationUpdate - update (replace) the properties of a Location already in state.
+- LocationDelete - remove a Location from state.
+ 
+_*Disclaimer: LocationDelete is a potentially hazardous operation and needs to 
+be done with care._
+
+
+## Permissions
+[permissions]: #permissions
+
+Creation of GS1 locations is restricted to agents acting on behalf of the 
+organization that is associated with the GS1 company prefix encoded in the GLN. 
+An organization must have the GS1 company prefix in the �gs1_company_prefixes� 
+metadata field for its Pike organization. (Organization-level metadata will need
+to be implemented in Pike.) When a location is created, its owning organization 
+is stored with the location in an �owner� field.
+
+Updates to locations are restricted to agents acting on behalf of the 
+organization stored in the location�s owner field. Only property fields can be 
+updated. They are updated by providing a complete and updated list of the 
+properties, and overwrites the entire list of property values. Location_id, 
+location_namespace, and owner fields are immutable.
+
+Deletion of locations is restricted to agents acting on behalf of the 
+organization in the location�s owner field. A setting will turn off deletion 
+entirely, with the intent that it could be enabled/disabled by a Grid 
+administrator. Disabling delete is useful because external systems may have 
+references to these locations and deleting them could leave dangling references.
+
+
+## GLN Validation
+[gln validation]: #glnvalidation
+
+Creation of a reusable GLN validation function to programmatically express the 
+equation used to validate a GLN. It validates GLN format to avoid mistype errors
+similar to a credit card validation. It's implemented as an extensible function 
+such that further validation steps can be added, if needed. For details on the 
+equation see: [Check digit validation](https://www.gs1us.org/tools/check-digit-calculator "Check digit validation")
```

Comment:
Suggest rewording this to state the intent instead of the mechanism. Perhaps:

GLN validation can reduce the opportunity for invalid data to enter the system. At a minimum, when GLNs entered in clients and later processed in smart contracts, they will need to conform to the GLN format and the check digit should be correct. See [Check digit calculator](https://www.gs1us.org/tools/check-digit-calculator "Check digit calculator").